subsys/bluetooth/host/Kconfig

```diff
@@ -509,6 +509,14 @@ config BT_KEYS_SAVE_AGING_COUNTER_ON_PAIRING
 	  time a successful pairing occurs. This increases flash wear out but offers
 	  a more correct finding of the oldest unused pairing info.
 
+config BT_SMP_MIN_ENC_KEY_SIZE
+	int "Minimum encryption key size accepted in octets"
+	range 7 16
+	range 16 16 if BT_SMP_SC_ONLY
+	default 7
```

Comment:
I think you need to make the default also conditional, so add a second line with `default 16 if BT_SMP_SC_ONLY`. I suspect you might get some warnings otherwise since the default would be pointing outside of the valid range.

Reply:
Actually, you could even make the prompt conditional to `BT_SMP_SC_ONLY` so that this is not visible to the user in that case.

Reply:
I like the conditional prompt solution :)

Reply:
Yep, I think that's the cleanest solution. I might be wrong, but I also think that if there's no prompt then the value isn't changeable through the app's prj.conf either, meaning you'd always get the default value. In that case you wouldn't need to add a separate range definition.